TypeCobol.Analysis/Report/ZCallPgmReport.cs

```diff
@@ -0,0 +1,402 @@
+        {
+            switch (up.Instruction.CodeElement.Type)
+            {
+                case CodeElementType.CallStatement:
+                    {
+                        CallStatement callStatement = up.Instruction.CodeElement as CallStatement;
+                        string pgmVar = callStatement?.InputParameters.First().StorageAreaOrValue.MainSymbolReference.Name;
+                        if (up.Variable.Name.Equals(pgmVar, StringComparison.InvariantCultureIgnoreCase))
+                        {
+                            var target = up.Instruction.CodeElement.CallSites.First().CallTarget;
+                            foreach (var callStyle in AlternativeCallList)
+                            {
+                                if (target != null && target.ToString().Equals(callStyle, StringComparison.OrdinalIgnoreCase))
+                                {
+                                    CallUsePoints.Add(up);
+                                    if (up.Variable.Type.Tag == Compiler.Types.Type.Tags.Group)
+                                    {
+                                        //Compute the Map of Level88 variable to their parent..
+                                        if (Level88SymbolParentSymbolMap == null)
+                                        {
+                                            Level88SymbolParentSymbolMap = new Dictionary<Symbol, Symbol>();
+                                        }
+                                        Level88SymbolCollector l88c = new Level88SymbolCollector(up.Variable);
+                                        up.Variable.Accept(l88c, Level88SymbolParentSymbolMap);
+                                    }
+                                    break;
+                                }
+                            }
+                        }
+                    }
+                    break;
+            }
+        }
+
+        /// <summary>
+        /// Called when a DefPoint is built. If the Def point is a level 88 variable of the target CALL variable
+        /// then we deceive DFA algorithm to think that the target variable of the DefPoint is the CALL variable
+        /// thus we change the target variable of the DefPoint.
+        /// </summary>
+        /// <param name="dfaBuilder">The Dfa Builder</param>
+        /// <param name="dp">The Def Point</param>
+        private void OnDefPoint(DataFlowGraphBuilder<Node, Symbol> dfaBuilder, DfaDefPoint<Node, Symbol> dp)
+        {
+            VariableSymbol sym = (VariableSymbol)dp.Variable;
+            Symbol parent = null;
+            if (sym.Level == 88 && sym.Value != null && Level88SymbolParentSymbolMap != null && Level88SymbolParentSymbolMap.TryGetValue(sym, out parent))
+            {//Now say that it is the parent variable which is the target of the definition.
+                dp.Variable = parent;
+                dp.UserData = sym;
+            }
+        }
```

Comment:
No this is a hack here, to make DFA think that when there is a level 88 variable assignment, it is it's parent variable which is assigned, DFA must track by the CALL the parent variable value changing and not the level 88 children variables.

Reply:
I agree, all I say is that this logic should be moved to `DefaultDataFlowGraphBuilder` otherwise all reports or class that would use the DFA results will have to perform this hack on their own, it must be factorized.

Reply:
Hum... I think it is a contextual logic, not sure that factorizinf such logic, will lead to a generic pattern.